pkg/entity/period.go

```diff
@@ -0,0 +1,120 @@
+package entity
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"time"
+
+	"github.com/SigNoz/signoz-otel-collector/pkg/errors"
+)
+
+// Period represents a value object representing specific time periods,
+// such as second or minute.
+// It implements the driver.Valuer and sql.Scanner interfaces for
+// working with databases,and the json.Marshaler and json.Unmarshaler
+// interfaces for JSON serialization and deserialization.
+type Period struct {
+	name string
+}
+
+var (
+	PeriodSecond = Period{name: "second"}
+	PeriodMinute = Period{name: "minute"}
```

Comment:
I don't see any practical use case where users want to limit the count/size every second/minute basis.